pkg/genericcli/context.go

```diff
@@ -0,0 +1,682 @@
+package genericcli
+
+import (
+	"cmp"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"slices"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/metal-stack/metal-lib/pkg/genericcli/printers"
+	"github.com/metal-stack/metal-lib/pkg/multisort"
+	"github.com/metal-stack/metal-lib/pkg/pointer"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	KeyName           = "name"
+	KeyAPIURL         = "api-url"
+	KeyAPIToken       = "api-token"
+	KeyDefaultProject = "default-project"
+	KeyTimeout        = "timeout"
+	KeyActivate       = "activate"
+	KeyProvider       = "provider"
+	KeyConfig         = "config"
+
+	defaultConfigName = "config.yaml"
+)
+
+var (
+	// errorNotImplemented for all functions that are not implemented yet
+	errorNotImplemented = fmt.Errorf("not implemented yet")
+)
+
+// Contexts contains all configuration Contexts
+type Contexts struct {
```

Comment:
I would consider this an internal representation, it should also be private and the consumers should not have an idea how the file is written or even where the information is stored. For what reason does it need to be public?